src/rgw/driver/sfs/types.cc

```diff
@@ -300,25 +383,37 @@ void Bucket::_undelete_object(ObjectRef objref, const rgw_obj_key & key,
   }
 }
 
+std::optional<ObjectRef> Bucket::get_from_db(const std::string& name) {
+  sqlite::SQLiteObjects objs(store->db_conn);
+  auto obj = objs.get_object(info.bucket.bucket_id, name);
+  if (!obj) {
+    return nullptr;
```

Comment:
the docs [here](https://en.cppreference.com/w/cpp/utility/optional) seem to suggest the right return here would be a `nullopt` instead of a `nullptr`. Which makes sense, given we're returning an `std::optional`. I suspect it may be implicitly converted, but implicit conversions are the devil.

Reply:
Fixed. The methods now return a pointer not std::optional